src/coreclr/vm/dllimport.cpp

```diff
@@ -2076,21 +2076,11 @@ void NDirectStubLinker::DoNDirect(ILCodeStream *pcsEmit, DWORD dwStubFlags, Meth
 
     if (SF_IsForwardStub(dwStubFlags)) // managed-to-native
     {
-
         if (SF_IsDelegateStub(dwStubFlags)) // delegate invocation
         {
             // get the delegate unmanaged target - we call a helper instead of just grabbing
             // the _methodPtrAux field because we may need to intercept the call for host, etc.
             pcsEmit->EmitLoadThis();
-#ifdef TARGET_64BIT
-            // on AMD64 GetDelegateTarget will return address of the generic stub for host when we are hosted
-            // and update the secret argument with real target - the secret arg will be embedded in the
-            // InlinedCallFrame by the JIT and fetched via TLS->Thread->Frame->Datum by the stub for host
-            pcsEmit->EmitCALL(METHOD__STUBHELPERS__GET_STUB_CONTEXT_ADDR, 0, 1);
-#else // !TARGET_64BIT
-            // we don't need to do this on x86 because stub for host is generated dynamically per target
-            pcsEmit->EmitLDNULL();
-#endif // !TARGET_64BIT
             pcsEmit->EmitCALL(METHOD__STUBHELPERS__GET_DELEGATE_TARGET, 2, 1);
```

Comment:
```suggestion
            pcsEmit->EmitCALL(METHOD__STUBHELPERS__GET_DELEGATE_TARGET, 1, 1);
```

Reply:
I think this is causing the CI test failures.

Also, `GET_DELEGATE_TARGET` signature in corelib.h should be updated to `SM_Delegate_RetIntPtr`, and the extra argument for `GetDelegateTarget` should be deleted in `StubHelpers.cs`.

Reply:
> `GetDelegateTarget` should be deleted in `StubHelpers.cs`.

Is `pThisUNSAFE` the extra argument that should be deleted?

Reply:
`ppStubArg` was previously deleted. I think it was about that. Testing the rest.